core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SnapshotApiServiceSpec.scala

```diff
@@ -155,41 +155,51 @@ class SnapshotApiServiceSpec extends ApiServiceSpec {
       check { assertResult(StatusCodes.NotFound) {status} }
   }
 
-  it should "return 200 when a user lists all workspaces" in withTestDataApiServicesAndUser(testData.userReader.userEmail.value) { services =>
-    Get(s"${testData.wsName.path}/snapshots", httpJson(
-      EnumerateSnapshotRequestBody(
-        offset = 0,
-        limit = 10
+  it should "return 200 when a user lists all snapshots in a workspace" in withTestDataApiServices { services =>
+    // First, create two workspaces
+    Post(s"${testData.wsName.path}/snapshots", httpJson(
+      DataRepoSnapshot(
+        name = "foo",
+        snapshotId = "realsnapshot"
       )
     )) ~>
       sealRoute(services.snapshotRoutes) ~>
       check {
-        val response = responseAs[DataRepoSnapshotList]
-        assertResult(StatusCodes.OK) {status}
-        assert(response.snapshots.size == 0)
+        val response = responseAs[DataRepoSnapshotReference]
+        assertResult(StatusCodes.Created) {status}
+        Post(s"${testData.wsName.path}/snapshots", httpJson(
+          DataRepoSnapshot(
+            name = "bar",
+            snapshotId = "realsnapshot2"
+          )
+        )) ~>
+          sealRoute(services.snapshotRoutes) ~>
+          check {
+            val response = responseAs[DataRepoSnapshotReference]
+            assertResult(StatusCodes.Created) {status}
+            // Then, list them both
+            Get(s"${testData.wsName.path}/snapshots?offset=0&limit=10") ~>
+              sealRoute(services.snapshotRoutes) ~>
+              check {
+                val response = responseAs[DataRepoSnapshotList]
+                assertResult(StatusCodes.OK) {status}
+                assert(response.snapshots.size == 2)
```

Comment:
for deeper testing, you could do:
```
                 response.snapshots should contain theSameElementsAs List(
                  DataRepoSnapshot(
                    name = "foo",
                    snapshotId = "realsnapshot"
                  ),
                  DataRepoSnapshot(
                    name = "bar",
                    snapshotId = "realsnapshot2"
                  )
                )
```
(untested/uncompiled code, could be wrong)

Reply:
One issue - the returned list is DataRepoSnapshotReferences, which have a lot more fields than the request model (some of which are generated by WM).
I verified the names from the inputs, though the snapshotIds are a little trickier since it requires parsing json.

Reply:
ah of course you're right it's a different return object!

For verifying the names, how about:
```
assertResult(Set("foo", "bar")) { response.snapshots.map(_.name).toSet }
```